python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+
+    relWeight = pexConfig.Field(
+        dtype=float, default=0.5,
+        doc="""Relative weighting of pre-subtraction images (higher -> greater influence of pre-sub.
+        images on fit)""")
+
+    tolerance = pexConfig.Field(
+        dtype=float, default=1e-7,
+        doc="Fit tolerance")
+
+    fitBgGradient = pexConfig.Field(
+        dtype=bool, default=True,
+        doc="Include parameters to fit for linear gradient in pre-sub. images")
+
+    fitSeparateNegParams = pexConfig.Field(
+        dtype=bool, default=False,
+        doc="Include parameters to fit for negative values (flux, gradient) separately from pos.")
+
+    # Config params for classification of detected diaSources as dipole or not
+    minSn = pexConfig.Field(
+        dtype=float, default=np.sqrt(2) * 5.0,
+        doc="Minimum quadrature sum of positive+negative lobe S/N to be considered a dipole")
+
+    maxFluxRatio = pexConfig.Field(
+        dtype=float, default=0.65,
+        doc="Maximum flux ratio in either lobe to be considered a dipole")
+
+    # Choose a maxChi2DoF corresponding to a significance level of at most 0.05
+    # (note this is actually a significance not a chi2 number)
+    maxChi2DoF = pexConfig.Field(
+        dtype=float, default=0.05,
+        doc="Maximum Chi2/DoF significance of fit to be considered a dipole.")
+
+    verbose = pexConfig.Field(
+        dtype=bool, default=False,
+        doc="Be verbose; this is slow")
+
+
+class DipoleFitTask(measBase.SingleFrameMeasurementTask):
+    """This is a subclass of SingleFrameMeasurementTask which can accept
+    three input images in its run() method. Because it subclasses
+    SingleFrameMeasurementTask, and calls
+    SingleFrameMeasurementTask.run() from its run() method, it still
+    can be used identically to a standard SingleFrameMeasurementTask.
+    """
+
+    ConfigClass = DipoleFitPluginConfig
+    _DefaultName = "ip_diffim_DipoleFit"
+
+    def __init__(self, schema, algMetadata=None, dpFitPluginConfig=None, **kwds):
+
+        measBase.SingleFrameMeasurementTask.__init__(self, schema, algMetadata, **kwds)
+
+        if dpFitPluginConfig is None:
+            dpFitPluginConfig = DipoleFitPluginConfig()
+        self.dipoleFitter = DipoleFitPlugin(dpFitPluginConfig, name=self._DefaultName,
+                                            schema=schema, metadata=algMetadata)
+
+    def run(self, sources, exposure, posImage=None, negImage=None, **kwds):
+        """!Run dipole measurement and classification
+
+        @param sources       diaSources that will be measured using dipole measurement
+        @param exposure      Exposure on which the diaSources were detected
+        @param posImage      "Positive" exposure from which the template was subtracted
+        @param negImage      "Negative" exposure which was subtracted from the posImage
+        @param **kwds        Sent to SingleFrameMeasurementTask
```

Comment:
These are all good suggestions. I have made the changes you suggested. I added the following to the docstring:
```
        @notes When posExp is None, will compute posImage = exposure + negExp.
        Likewise, when negExp is None, will compute negImage = posExp - exposure.
        If both posExp and negExp are none, will attempt to fit the dipole to just the exposure
        with no constraint.
```
I have also updated the code so that it accepts None for either posExp, negExp, or both. I have also added testCases to test these and validated that they pass.